drivers/adc/adc_stm32.c

```diff
@@ -1562,20 +1515,62 @@ static void adc_stm32_cfg_func_##index(void){ EMPTY }
 
 #else /* CONFIG_ADC_STM32_DMA */
 
-#define ADC_STM32_IRQ_CONFIG(index)					\
-static void adc_stm32_cfg_func_##index(void)				\
-{									\
-	IRQ_CONNECT(DT_INST_IRQN(index),				\
-		    DT_INST_IRQ(index, priority),			\
-		    adc_stm32_isr, DEVICE_DT_INST_GET(index), 0);	\
-	irq_enable(DT_INST_IRQN(index));				\
+static bool adc_stm32_is_irq_active(ADC_TypeDef *adc)
+{
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f4_adc)
+	return LL_ADC_IsActiveFlag_EOCS(adc) ||
+#else
+	return LL_ADC_IsActiveFlag_EOC(adc) ||
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(st_stm32f4_adc) */
+	       LL_ADC_IsActiveFlag_OVR(adc) || LL_ADC_IsActiveFlag_JEOS(adc) ||
+	       LL_ADC_IsActiveFlag_AWD1(adc);
 }
-#define ADC_STM32_IRQ_FUNC(index)					\
-	.irq_cfg_func = adc_stm32_cfg_func_##index,
-#define ADC_DMA_CHANNEL(id, dir, DIR, src, dest)
 
-#endif /* CONFIG_ADC_STM32_DMA && CONFIG_ADC_STM32_SHARED_IRQS */
+#define FIRST_WITH_IRQN_INTERNAL(index, irqn)                                                      \
+	COND_CODE_1(IS_EQ(irqn, DT_INST_IRQN(index)), (index, ), (EMPTY, ))
```

Comment:
you will have to format these manually

```suggestion
	COND_CODE_1(IS_EQ(irqn, DT_INST_IRQN(index)), (index,), (EMPTY,))
```

Reply:
Done!